Examples/src/components/Examples/Charts2D/Animations/GenericAnimation/GENERATED_SRC.ts

```diff
@@ -0,0 +1,244 @@
+        },
+        onCompleted: () => {
+            animationSinPoint.reset();
+        }
+    });
+
+    const animationCosPoint = new GenericAnimation<ICustomAnnotationOptions>({
+        from: { x1: data.xValues[0], y1: data.y1Values[0] },
+        to: { x1: data.xValues[data.xValues.length - 1], y1: data.y1Values[data.y1Values.length - 1] },
+        duration: ANIMATION_DURATION,
+        ease: easing.inOutSine,
+        onAnimate: (from, to, progress) => {
+            const point = Math.floor(data.xValues.length * progress);
+            annotations.pointCosAnnotation.x1 = data.xValues[point];
+            annotations.pointCosAnnotation.y1 = data.y1Values[point];
+        },
+        onCompleted: () => {
+            animationCosPoint.reset();
+        }
+    });
+
+    return { animation, animationSinPoint, animationCosPoint };
+};
+
+let scs: SciChartSurface;
+let animation: GenericAnimation<ILineAnnotationOptions>;
+let animationSinPoint: GenericAnimation<ICustomAnnotationOptions>;
+let animationCosPoint: GenericAnimation<ICustomAnnotationOptions>;
+let startAnimation: () => void;
+
+// React component needed as our examples app is react.
+// SciChart can be used in Angular, Vue, Blazor and vanilla JS! See our Github repo for more info
+export default function GenericAnimations() {
+    const [isStarted, setStarted] = React.useState<boolean>(false);
+    const [isCanceled, setCanceled] = React.useState<boolean>(false);
+    React.useEffect(() => {
+        (async () => {
+            const res = await drawExample();
+            scs = res.sciChartSurface;
+            animation = res.animations.animation;
+            animationSinPoint = res.animations.animationSinPoint;
+            animationCosPoint = res.animations.animationCosPoint;
+            startAnimation = res.startAnimation;
+        })();
+        // Delete sciChartSurface on unmount component to prevent memory leak
+        return () => scs?.delete();
+    }, []);
+
+    const handleStartClick = () => {
+        setStarted(true);
+        startAnimation();
+    }
+
+    const handleRestartClick = () => {
+        cancelAnimations();
+        startAnimation();
+        animation?.reset();
+        animationCosPoint?.reset();
+        animationSinPoint?.reset();
+        setStarted(true);
+        setCanceled(false);
+    };
+
+    const handleCancelClick = () => {
+        cancelAnimations();
+        setStarted(false);
+        setCanceled(true);
+    };
+
+    const cancelAnimations = () => {
+        animation?.cancel();
+        animationCosPoint?.cancel();
+        animationSinPoint?.cancel();
+    }
+
+    return (
+        <>
+            <div id={divElementId} className={classes.ChartWrapper} />
+            <div className={classes.ButtonsWrapper}>
+                <Button className={classes.ButtonsText} size="medium" onClick={handleStartClick} disabled={isStarted || isCanceled}>
+                    start
```

Comment:
Why do we need to have three button? Why not just "Start" and "Stop"?

Reply:
This can confuse a bit, because we don't have like a "pause" function, we just canceling through the method, so in that case we need to recreate the animations, basically, buttons cancel and restart works as you wrote, but in this case "restart" more better wording from tech perspective =/ If you have restart and cancel it should be started at some point, so I think "Start" button can be there